temporal-sdk/src/main/java/io/temporal/failure/DefaultFailureConverter.java

```diff
@@ -189,7 +189,14 @@ private TemporalFailure failureToExceptionImpl(Failure failure, DataConverter da
         }
       case FAILUREINFO_NOT_SET:
       default:
-        throw new IllegalArgumentException("Failure info not set");
+        // All unknown types are considered to be retryable ApplicationError.
```

Comment:
I checked all other SDKs and the Java SDK was the only SDK that was throwing an exception here instead of converting to an application failure